File: java-example/src/test/java/ru/stqa/training/selenium/pages/AdminPanelLoginPage.java
package ru.stqa.training.selenium.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class AdminPanelLoginPage extends Page {

    public AdminPanelLoginPage(WebDriver driver) {
        super(driver);
    }

    public AdminPanelLoginPage open() {
        driver.get("http://localhost/litecart/admin");
        return this;
    }

    public boolean isOnThisPage() {
        return driver.findElements(By.id("box-login")).size() > 0;
    }

    public AdminPanelLoginPage enterUsername(String username) {
        driver.findElement(By.name("username")).sendKeys(username);
        return this;
    }

    public AdminPanelLoginPage enterPassword(String password) {
        driver.findElement(By.name("password")).sendKeys(password);
        return this;
    }

    public void submitLogin() {
        driver.findElement(By.name("login")).click();
        wait.until((WebDriver d) -> d.findElement(By.id("box-apps-menu")));
    }

}

File: ruby-example/pages/admin_panel_login_page.rb
require 'selenium-webdriver'

class AdminPanelLoginPage

  def initialize(driver)
    @driver = driver
    @wait = Selenium::WebDriver::Wait.new(:timeout => 10) # seconds
  end

  def open
    @driver.get 'http://localhost/litecart/admin'
    self
  end

  def on_this_page?
    @driver.find_elements(:id => 'box-login').size > 0
  end

  def enter_username(username)
    @driver.find_element(:name => 'username').send_keys username
    self
  end

  def enter_password(password)
    @driver.find_element(:name => 'password').send_keys password
    self
  end

  def submit_login
    @driver.find_element(:name, 'login').click
    @wait.until { @driver.find_element :id => 'box-apps-menu' }
  end
end

